flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/topic/TopicPartition.java

```diff
@@ -96,8 +100,12 @@ public TopicPartition(String topic, List<TopicRange> ranges) {
     public TopicPartition(String topic, int partitionId, List<TopicRange> ranges) {
         checkArgument(partitionId >= NON_PARTITION_ID, "Invalid partition id.");
 
-        this.topic = topicName(checkNotNull(topic));
-        this.partitionId = partitionId;
+        TopicName topicName = TopicName.get(topic);
+        this.topic = topicName.getPartitionedTopicName();
+        this.partitionId =
+                partitionId == NON_PARTITION_ID && topicName.isPartitioned()
```

Comment:
If we forced replace the partitionId here, we may need to print a warn level log to avoid confuse.